66_HLSLBxDFTests/app_resources/tests_common.hlsl

```diff
@@ -0,0 +1,668 @@
+    {
+        SBxDFTestResources retval;
+        retval.rng = nbl::hlsl::Xoroshiro64Star::construct(seed);
+        nbl::hlsl::random::DimAdaptorRecursive<nbl::hlsl::Xoroshiro64Star, 2> rng_vec2 = nbl::hlsl::random::DimAdaptorRecursive<nbl::hlsl::Xoroshiro64Star, 2>::construct(retval.rng);
+        nbl::hlsl::random::DimAdaptorRecursive<nbl::hlsl::Xoroshiro64Star, 3> rng_vec3 = nbl::hlsl::random::DimAdaptorRecursive<nbl::hlsl::Xoroshiro64Star, 3>::construct(retval.rng);
+        retval.u = ConvertToFloat01<uint32_t3>::__call(rng_vec3());
+        retval.u.x = hlsl::clamp(retval.u.x, retval.eps, 1.f-retval.eps);
+        retval.u.y = hlsl::clamp(retval.u.y, retval.eps, 1.f-retval.eps);
+
+        retval.V.direction = nbl::hlsl::normalize<float32_t3>(sampling::UniformSphere<float>::generate(ConvertToFloat01<uint32_t2>::__call(rng_vec2())));
+        retval.N = nbl::hlsl::normalize<float32_t3>(sampling::UniformSphere<float>::generate(ConvertToFloat01<uint32_t2>::__call(rng_vec2())));
+        
+        float32_t3 tangent, bitangent;
+        math::frisvad<float32_t3>(retval.N, tangent, bitangent);
+        tangent = nbl::hlsl::normalize<float32_t3>(tangent);
+        bitangent = nbl::hlsl::normalize<float32_t3>(bitangent);
+
+        const float angle = 2.0f * numbers::pi<float> * ConvertToFloat01<uint32_t>::__call(retval.rng());
+        float32_t4x4 rot = math::linalg::promote_affine<4, 4>(math::linalg::rotation_mat(angle, retval.N));
+        retval.T = mul(rot, float32_t4(tangent,1)).xyz;
+        retval.B = mul(rot, float32_t4(bitangent,1)).xyz;
+
+        retval.alpha.x = ConvertToFloat01<uint32_t>::__call(retval.rng());
+        retval.alpha.y = ConvertToFloat01<uint32_t>::__call(retval.rng());
+        retval.eta = ConvertToFloat01<uint32_t2>::__call(rng_vec2()) * hlsl::promote<float32_t2>(1.5) + hlsl::promote<float32_t2>(1.1); // range [1.1,2.6], also only do eta = eta/1.0 (air)
+        retval.luma_coeff = float32_t3(0.2126, 0.7152, 0.0722); // luma coefficients for Rec. 709
+
+        retval.Dinc = ConvertToFloat01<uint32_t>::__call(retval.rng()) * 2400.0f + 100.0f;
+        retval.etaThinFilm = ConvertToFloat01<uint32_t>::__call(retval.rng()) * 0.5 + 1.1f; // range [1.1,1.6]
+        return retval;
+    }
+
+    float eps = 1e-3;   // epsilon
+    uint32_t halfSeed;     // init state seed, for debugging
+
+    nbl::hlsl::Xoroshiro64Star rng;
+    ray_dir_info_t V;
+    float32_t3 N;
+    float32_t3 T;
+    float32_t3 B;
+
+    float32_t3 u;
+    float32_t2 alpha;
+    float32_t2 eta; // (eta, etak)
+    float32_t3 luma_coeff;
+
+    // thin film stuff;
+    float Dinc; // in nm [100, 2500]
+    float etaThinFilm;
+};
+
+struct STestInitParams
+{
+    bool logInfo;
+    uint32_t halfSeed;
+    uint32_t samples;
+    uint32_t thetaSplits;
+    uint32_t phiSplits;
+    bool writeFrequencies;
+    bool immediateFail;
+    bool verbose;
+};
+
+enum ErrorType : uint32_t
+{
+    BET_NONE = 0,
```

Comment:
reanme to `TestResult` and only put Error in the name of things which are errors, also make sure non-error codes are non-negative
https://discord.com/channels/593902898015109131/1317142868326092870/1450425434985599046